src/guc.c

```diff
@@ -437,6 +438,26 @@ _guc_init(void)
 							 NULL,
 							 NULL);
 
+	/*
+	 * Define the limit on number of invalidation-based refreshes we allow per
+	 * refresh call. If this limit is exceeded, fall back to a single refresh that
+	 * covers the range decided by the min and max invalidated time.
+	 */
+	DefineCustomIntVariable(MAKE_EXTOPTION("materializations_per_refresh_window"),
+							"Max number of materializations per cagg refresh window",
+							"The maximal number of individual refreshes per cagg refresh. If more "
+							"refreshes need to be performed, they are merged into a larger "
+							"single refresh.",
+							&ts_guc_cagg_max_individual_materializations,
+							10,
+							0,
+							INT_MAX,
+							PGC_SUSET,
```

Comment:
Wondering if we should switch this to `PGC_USERSET`, what do you think @fabriziomello?

Reply:
Ahhh... very good point... I guess this should be `PGC_USERSET` to make it easier for users and even for us testing it on cloud using `tsdbadmin`. Can't see any problem relaxing this restriction!

Reply:
Ok, I will change it before merging.